homeassistant/components/rabbitair/fan.py

```diff
@@ -0,0 +1,148 @@
+        elif self._is_model(Model.A3):
+            # A3 does not support Pollen mode
+            self._attr_preset_modes = [
+                k for k in PRESET_MODES if k != PRESET_MODE_POLLEN
+            ]
+
+        self._attr_speed_count = len(SPEED_LIST)
+
+        self._update_state()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_state()
+        super()._handle_coordinator_update()
+
+    def _update_state(self) -> None:
+        data = self.coordinator.data
+
+        # Power on/off
+        self._power = data.power
+
+        # Speed as a percentage
+        if data.speed is None:
+            self._attr_percentage = None
+        elif data.speed is Speed.SuperSilent:
+            self._attr_percentage = 0
+        else:
+            self._attr_percentage = ordered_list_item_to_percentage(
+                SPEED_LIST, data.speed
+            )
+
+        # Preset mode
+        if data.mode is None:
+            self._attr_preset_mode = None
+        else:
+            # Get key by value in dictionary
+            self._attr_preset_mode = next(
+                k for k, v in PRESET_MODES.items() if v == data.mode
+            )
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self._set_state(mode=PRESET_MODES[preset_mode])
+        self._attr_preset_mode = preset_mode
+        self.async_write_ha_state()
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the speed of the fan, as a percentage."""
+        value = percentage_to_ordered_list_item(SPEED_LIST, percentage)
+        await self._set_state(speed=value)
+        self._attr_percentage = percentage
+        self.async_write_ha_state()
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn on the fan."""
+        mode_value: Mode | None = None
+        if preset_mode is not None:
+            mode_value = PRESET_MODES[preset_mode]
+        speed_value: Speed | None = None
+        if percentage is not None:
+            speed_value = percentage_to_ordered_list_item(SPEED_LIST, percentage)
+        await self._set_state(power=True, mode=mode_value, speed=speed_value)
+        self._power = True
+        if percentage is not None:
+            self._attr_percentage = percentage
+        if preset_mode is not None:
+            self._attr_preset_mode = preset_mode
+        self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the fan off."""
+        await self._set_state(power=False)
+        self._power = False
+        self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if device is on."""
+        return self._power
```

Comment:
This property can be removed, and replace `self._power` with `self._attr_is_on` that is already available.

Reply:
Unfortunately, that's not the case. The parent class FanEntity overrides this method so that it does not use the value of the _attr_is_on field. But we can override it again to get the previous behavior back.